### PR TITLE
[CI:DOCS] `CONTRIBUTING.md`: Update contact information

### LINKED ISSUE #4308: CONTRIBUTING.md out of date? Freenode?
The `CONTRIBUTING.md` file directs new contributors to the `#buildah` channel on `irc.freenode.net`, which appears to be dead. It also appears there may have been some major changes and drama at freenode in the last year that resulted in many projects and individuals moving to libera.chat, or other platforms.

Is `CONTRIBUTING.md` out of date? What are the current recommendations for communication? Where should more general questions that aren't direct issues with this repository or software go? Such as if someone had questions on how to contribute, standards, or similar? Or is it just that no one has the time or interest for that relating to this project?

### PR DESCRIPTION
**Pull request by AndASM**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:
`CONTRIBUTING.md` is outdated. New contributors will be directed to a dead freenode channel.

Changes:
- Freenode is no longer used, IRC has moved to [libera.chat](https://libera.chat).
- \[matrix\] is federated into the `#podman` channel and can be used instead of IRC.

#### How to verify it
Read `CONTRIBUTING.md`.

#### Which issue(s) this PR fixes:

Fixes #4308

#### Special notes for your reviewer:
I am assuming this counts as "developer-facing" not "user-facing" so I have put `None` under the release notes. Is that correct?

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@TomSweeneyRedHat PTAL

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -143,14 +143,11 @@ is available, showing all PRs awaiting review and approval.
...
 
 For general questions or discussions, please use the
-IRC group on `irc.freenode.net` called `buildah`
-that has been setup.
+IRC channel `#podman` on `irc.libera.chat`. If you are unfamiliar with IRC you can start a web client at https://web.libera.chat/#podman.
 
-### For discussions around issues/bugs and features:
-
-#### Buildah Mailing List
+Alternatively [\[matrix\]](https://matrix.org) can be used to access the same channel via federiation at https://matrix.to/#/#podman:chat.fedoraproject.org.
```
Comment (TomSweeneyRedHat) [inline CONTRIBUTING.md:148]: I'm 99% sure it should be federation and not federation.

```suggestion
Alternatively, [\[matrix\]](https://matrix.org) can be used to access the same channel via federation at https://matrix.to/#/#podman:chat.fedoraproject.org.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -159,6 +156,12 @@ and
 [PRs](https://github.com/containers/buildah/pulls)
 tracking system.
 
+#### Buildah Mailing List
+
+_This list sees low traffic, the GitHub links or IRC above are better options in most cases._
```
Comment (TomSweeneyRedHat) [inline CONTRIBUTING.md:161]:
```suggestion
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -143,14 +143,11 @@ is available, showing all PRs awaiting review and approval.
...
-### For discussions around issues/bugs and features:
-
-#### Buildah Mailing List
+Alternatively [\[matrix\]](https://matrix.org) can be used to access the same channel via federiation at https://matrix.to/#/#podman:chat.fedoraproject.org.
 
-You can join the Buildah mailing list by sending an email to `buildah-join@lists.buildah.io` with the word `subscribe` in the subject.  You can also go to this [page](https://lists.podman.io/admin/lists/buildah.lists.buildah.io/), then scroll down to the bottom of the page and enter your email and optionally name, then click on the "Subscribe" button.
+### For discussions around issues/bugs and features:
 
 #### GitHub
 You can also use the github
```
Comment (TomSweeneyRedHat) [inline CONTRIBUTING.md:153]: I know it's old, but please touch up
```suggestion
You can also use GitHub
```

Comment (TomSweeneyRedHat) [inline CONTRIBUTING.md:153]: Also, please add:
`[Discussions](https://github.com/containers/podman/discussions),`

**Comment by rhatdan** [comment]
@AndASM Are you still working on this, or would you rather us finish it?

**Comment by AndASM** [comment]
> @AndASM Are you still working on this, or would you rather us finish it?

Sorry about that, I missed the update emails and have been distracted and busy lately. I'll try to take a look tomorrow if you haven't otherwise taken care of it.

**Review by flouthoc — CHANGES_REQUESTED** [review]
Could you reduce length of your commit title please ?

**Comment by flouthoc** [comment]
Oops it was commit  2c06b50cf1635f1504bf83a19fb414a6f8bc99c5 caused by using github rebase , @AndASM Could you squash commits into one ?

**Comment by rhatdan** [comment]
@AndASM Are you still working on this one?

**Comment by AndASM** [comment]
Oops, sorry about that! Life has been a bit crazy of late.

I squashed the commits into one, and rebased it on the current main branch. Is this still okay to pull? It might help someone to have those updates to the documentation. @rhatdan?

**Comment by rhatdan** [comment]
I can't seem to reopen, could you open a new PR with just one commit.

**Pull request closed without merging**